all.yml

```
@@ -11,6 +11,14 @@
     include_tasks: tasks/certificate_authority.yml
     when: (self_signed|bool and regenerate_ca|bool and kafka_broker_listeners | ssl_required(ssl_enabled)) or create_mds_certs|bool
 
+  - name: Save Security File - Secrets Protection
+    set_fact: secrets_protection_security_file={{ lookup('file', secrets_protection_security_file) }}
+    when: (secrets_protection_masterkey|length>0) and (secrets_protection_security_file|length>0)
+
+  - name: Generate Master Encryption Key - Secrets Protection
+    include_tasks: tasks/masterkey.yml
+    when: (secrets_protection_enabled|bool) and (not secrets_protection_masterkey)
```

```suggestion
  - name: Generate Master Encryption Key - Secrets Protection
    include_tasks: tasks/masterkey.yml
    when: secrets_protection_enabled|bool
```

please move all master key logic into tasks/masterkey.yml that way all.yml looks cleaner

---

you can have the same conditionals in the tasks file